scripts/common/results/result_info.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Structures for a variety of different test results."""
+
+
+def build_test_result(test_id,
+                      result,
+                      result_type='total_time',
+                      result_units='ms',
+                      test_harness='unknown',
+                      test_environment='unknown'):
+  """Core information about the results of the test.
+
+  Args:
+      test_id (str): Id when combined with test_source should represent a unique
```

Comment:
we don't specify types when describing parameters. if you want to enforce types, you will need to check user input values using isinstance().